### ISSUE TITLE
Allow querying of daemon's log-driver and log-opt via remote API

### ISSUE THREAD

**Issue by xbglowx** [docker/docker#25508]
Sorry for being brief, but I feel the subject says it all. If you need anything else, please let me know.

**Comment by cpuguy83** [comment]
@xbglowx This information is available on the container's `HostConfig` via `GET /containers/<id>/json`

**Comment by cpuguy83** [comment]
Closing since this should already be available, but please feel free to provide more detail of what you need if this is not enough.
Thanks!

**Comment by xbglowx** [comment]
@cpuguy83 you are correct that it is available via the container, but what happens if someone creates a container that overrides the daemon options. I think there is value for being able to query the log driver/options of the daemon, don't you?

**Comment by xbglowx** [comment]
Sorry, just saw your second comment. A little more info:

Let's say some type configuration management is responsible for running containers and making sure that when there is any changes, the older container is replaced with a new one. If configuration management can't read the default logging options, which come from the docker daemon, how will it know if there is a change/diff?

**Comment by xbglowx** [comment]
I just thought of one hack that could be a work around:

1. Have configuration management launch a new temp container using the new config, which in this case, would have nothing set for log_config
1. Inspect the container to see contents of log_config
1. Delete temp container
1. Compare output of log_config between temp container and the other container that we care about, to see if any action is required
.
The above should work, but it is just a work around.

**Comment by cpuguy83** [comment]
Wouldn't the configuration management be managing the docker daemon configuration as well?

**Comment by xbglowx** [comment]
That is a good point, but, it is not guaranteed. For those that have decided to not let their configuration management control the daemon, for whatever reason, will run into problems.

Regardless of how they are running their daemon, querying the API to get the information will work in either scenario. Plus configuration managements like Ansible and Saltstack interact with the docker with help from the Python module docker-py. It seems natural to continue on that path and use it to also query for the log options.

**Comment by joegross** [comment]
I love burgers and I'm particular about my fixings. The good burger joints put their default fixings on the menu but others force you to ask first.

The worst ones won't tell you their defaults and force you to specify everything to keep 'em from getting it wrong. It's especially annoying if you hate pickles but forget to be explicit when you place your order.

**Comment by cpuguy83** [comment]
I agree it would be helpful to be able to know how the daemon is configured.

**Comment by yongtang** [comment]
The `log-driver` is already available in `GET /info` and `docker info` through field `LoggingDriver`:
```
ubuntu@ubuntu:~/docker$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 55
Server Version: 1.12.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 41
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
```

Though `log-opt` is not there in `GET /info` / `docker info` yet.

**Comment by yongtang** [comment]
Created a PR #27422 to add a `LoggingConfig` field in `types.Info` for `docker info` and `GET /info`. Please take a look and let me know if there are any issues.

**Comment by thaJeztah** [comment]
Also see https://github.com/docker/docker/issues/21559, for getting the active daemon configuration "in general"

**Comment by thaJeztah** [comment]
I see the PR was closed because there were concerns about potentially leaking sensitive config-options in `docker info`, which might still be a valid concern (but _could_ be solvable, as long as we would make it the responsibility of the logging-drives to provide the information to show (as they _should_ know which fields need to be sanitized).


Considering if we should close this one in favour of https://github.com/moby/moby/issues/21559, or if we can maintain this one as a smaller subset of that work 🤔
